docs(application): application for web3 open jurisdiction stack (#1834)
w3f/Grants-Program: applications/web3-open-jurisdiction-stack.md

Additionally, the project includes developing a technical architecture consisting of pallets and smart contracts for the open jurisdiction stack that we envision taking the form of a system parachain.

As part of the project, a proof of concept (POC) for one of the pallets that should be included in the open jurisdiction stack will also be developed.

The objective of this research and architecture is to create a solution that guarantees its adoption and usefulness. Our future proposals will focus on developing and delivering the stack for community use.

**The problem(s) that you want to investigate, and why these are important.**

Every community striving for scalability must possess a mechanism to effectively address and resolve disagreements. By establishing a framework to handle such conflicts, the system can exhibit resilience and withstand challenges that may arise when people and entities do not agree and may fall apart.

> There is a glaring concrete example that highlights the issue of treasury management. When a grant is released by the Web3 Foundation, Polkadot, or Kusama Treasury, there is currently no recourse if the recipient fails to deliver or if there is a disagreement between the issuer and recipient regarding KPIs or Service Level Agreements. This problem is just one of many within the same realm and has raised serious concerns in the community for months (as an example, read this [thread](https://twitter.com/0xgoku_/status/1644260018659655680?s=20)). Despite this, as of today, there is still no solution in place to address these issues.

Reviewer: I would disagree with this first statement, as the Web3 Foundation Grants Program is off-chain and therefore a different process. Paying grantees per milestone helps to avoid this.

Reviewer: Thanks @0xlyd what I was getting at here is that I don't think you can lump them all into one category. W3F grant funds don't come from the treasury, unlike treasury proposals and bounties. Therefore a "grant released by the Web3 Foundation" doesn't fall under treasury management. It's a separate source of funding altogether. 

The majority of your proposal focuses on the treasury, and personally this is where I'd like to see the focus of this research. I definitely think these are needed tools for the treasury, but I'm not so sure when it comes to the W3F grants program. The long term goal is to move the grants program entirely on-chain, but until that happens, the grants committee will remain in control and some trust in W3F will be needed.

I'm no legal expert, but I do agree with the overall need for decentralized arbitration. I hope we can eventually further decentralize the grants program, similar to how the Polkadot council was dissolved and control was handed over to the larger fellowship and community.

For now, I'd love to see a PoC proposed and built for the treasury, so that when it does come time to move on-chain, we can utilize it.

Author: @keeganquigley thanks for expanding your point, appreciated 🙏 .

I can see it better now from W3F point's of view and why you would like us to focus exclusively on the treasury.

The reason why we would still like to include grants in the scope of our research is because we feel that it may help in creating a more flexible stack for future needs. We can't be sure that W3F will eventually adopt 100% of the stack of the treasury or if it would need something customized.

What we would like to do (just for the architecture part, the POC would be a functioning component/module) is to envision a system that's flexible enough. If we do focus only on the current treasury needs it will absorb those requirements but may not be able to accommodate other needs.

If your counterpoint is "but those other needs will come from oranges not apples, don't worry about that and just deliver the best for apples" it's a fair argument. My thought on this is that eventually we are talking about fruits (to stay in the analogy 😛 ) and how work aims to be "generic enough", but if you **strongly** advise against this we will amend the proposal accordingly 🤝

Reviewer: Thanks @0xlyd for defending your position. Thanks also for your point on hybrid DAOs, as our ecosystem typically doesn't use tools like Snapshot for the most part since our governance is all on-chain. I also agree with your assessment that we currently can't be sure of what the future of the W3F grants program will look like.

I think as long as the distinctions are made clear, the research could overall be beneficial for the community.

By taking this approach, we can ensure that the tech stack addresses the actual needs of ecosystem actors and dynamics, maximising the chances of concrete adoption.

The reason why we think a thorough analysis phase (as per software engineering best practices) is needed in this case is because the impact of the software solution has a real world aspect to it aside from the involvement of several human stakeholders. We strongly advise against going heads down towards development as it is likely to be something of no use if not backed by data collected methodically and actual user interaction.

To give an idea, although the actual questionnaire will be part of the _Milestone 1-1_, the discussion guide will be around questions such as:

- Do you find the process of applying to grants fair and just?
- Do you feel protected in case issues do arise while working on a grant?
- Do you feel there should be a clear process for handling such issues?
- Do you feel comfortable in allocating funds to entities?
- Did it ever happen that you felt the outcome of a relationship with a treasury was unfair?
- Do you think there is a need for a legally binding resolution method?

> - How would you currently resolve a disagreement with your counterparty?

Reviewer: I want to clarify here that the W3F Grants Program is completely separate from treasury proposals. Can you specify whether you are referring to treasury proposals and bounties here?

Author: The discussion guide sample is regarding treasury proposals indeed 🙏  but just to be clear we would like to explore also grants flow of the Web3 Foundation (and others) to understand all the nuances.

Eventually the common abstraction between all these is the movement of funds from a common pot for a common good through a public initiative. We are interested in making this process as bulletproof as possible

Reviewer: Thanks @0xlyd perhaps this is just a matter of confusing terminology. For example, I wouldn't call a successful treasury proposal a "grant". Maybe "spending request" or "treasury funds" would be more appropriate. Being that the W3F grants program is currently off-chain, comparing it with treasury proposals is kind of like comparing apples & oranges at this point. For example we do have contractual agreements with grantees, as well as terms & conditions, a contributor license agreement (CLA), private grants, etc. Essentially our program has some legal frameworks that the treasury might not have. I think this might create different considerations for each.

The W3F grants program also has a smaller scope. We mainly fund software and research, whereas the treasury has an expanded scope and can fund other things such as events and education initiatives. I see the treasury as more of a decentralized, open pot of "public funds".

It may seem pedantic, but in my mind the terminology is very important, as it could potentially distort the data if the participants are confused as to what you are referring to. The complexity of our ecosystem requires the need for explicit terms. If you could update your application to distinguish the difference between the areas of research I think this would help to avoid confusion.

Reviewer: Thanks for your thorough answer @0xlyd sounds good! Feel free to ping me when I should take another look.

Additionally, the project includes developing a technical architecture consisting of pallets and smart contracts for the open jurisdiction stack that we envision taking the form of a system parachain.

As part of the project, a proof of concept (POC) for one of the pallets that should be included in the open jurisdiction stack will also be developed.

The objective of this research and architecture is to create a solution that guarantees its adoption and usefulness. Our future proposals will focus on developing and delivering the stack for community use.

**The problem(s) that you want to investigate, and why these are important.**

Every community striving for scalability must possess a mechanism to effectively address and resolve disagreements. By establishing a framework to handle such conflicts, the system can exhibit resilience and withstand challenges that may arise when people and entities do not agree and may fall apart.

> There is a glaring concrete example that highlights the issue of treasury management. When a grant is released by the Web3 Foundation, Polkadot, or Kusama Treasury, there is currently no recourse if the recipient fails to deliver or if there is a disagreement between the issuer and recipient regarding KPIs or Service Level Agreements. This problem is just one of many within the same realm and has raised serious concerns in the community for months (as an example, read this [thread](https://twitter.com/0xgoku_/status/1644260018659655680?s=20)). Despite this, as of today, there is still no solution in place with regards to treasuries and partially for Web3 Foundation grants to address these issues.

Reviewer 2: > When a grant is released by the Web3 Foundation, Polkadot, or Kusama Treasury, there is currently no recourse if the recipient fails to deliver or if there is a disagreement between the issuer and recipient regarding KPIs or Service Level Agreements.

As already [mentioned](https://github.com/w3f/Grants-Program/pull/1834#discussion_r1261616230) by @keeganquigley, there is actually a recourse mechanism within the W3F Grants' Program, as we do not support any upfront payment. I think clarifying this would help avoid any confusion.

> read this [thread](https://twitter.com/0xgoku_/status/1644260018659655680?s=20)

Very interesting read. I was just wondering if you engaged in this discussion as well? Alternatively, I was wondering if you could lead me to some resources where your team engaged in discussion with the community, for example on the polkadot forum or on X.

A sample discussion guide will be along the following lines:

Grants research

- Do you find the process of applying to grants fair and just?
- Do you feel protected in case issues do arise while working on a grant?
- Do you feel there should be a clear process for handling such issues?
- Are you comfortable with the degree of trust in the Web3 Foundation involved in applying for a grant?

The goal here will be to analyse in detail a minimum of 10 grants and engage with involved parties to understand how their experience was. The main focus being:

- How would you rate your experience?
- Did you feel you were dependent on your counterparty acting fairly?
- Did you ever consider that your counterparty may not fulfil their obligations (e.g. deliver work or send a payment)?
- How would you have behaved in such a situation?
- Do you think a clear procedure for handling disagreements would have helped?
- Do you think it would be better for such a procedure to have legal value in the real world?

**The data collection and analysis procedures.**

> To facilitate data collection and collaboration, treasury and grants data will be documented in a Google spreadsheet. This spreadsheet will predominantly contain publicly available data but will also include anonymized opinions from stakeholders regarding the effectiveness of specific grants.

Reviewer 2: I think it's a good idea to anonymize certain data here. Could you explain how this would be decided and who'd be the decision maker?

> Throughout the discussion guide we aim to leverage NPS and similar metrics to be able to extract readable graphs that can inform the subsequent development of the stack and also to be publicly available references for the community of the work done.

Reviewer 2: ```suggestion
Throughout the discussion guide we aim to leverage NPS (Net Promoter Score) and similar metrics to be able to extract readable graphs that can inform the subsequent development of the stack and also to be publicly available references for the community of the work done.
```

I just thought it'd be easier to read if this acronym would be explained at its first appearance (rather than 2nd).

**The expected results and how they would be double-checked by the grants team (reproducibility of the data analysis).**

> While we anticipate observing a higher level of dissatisfaction in proposals and grants that involve subjective KPIs, we will be judicious in selecting relevant interviewees, adhering to the principle of diminishing returns. Ensuring compliance with legal and ethical standards, we will consider utilising a variation of the Net Promoter Score (NPS) to gauge dissatisfaction with the outcomes.

Reviewer 2: ```suggestion
While we anticipate observing a higher level of dissatisfaction in proposals and grants that involve subjective KPIs, we will be judicious in selecting relevant interviewees, adhering to the principle of diminishing returns. Ensuring compliance with legal and ethical standards, we will consider utilising a variation of the NPS to gauge dissatisfaction with the outcomes.
```

> The community will have access to all recorded interviews, discussion guides, reports for each interview, spreadsheet with key metrics and graphs that plot the most significant data points from the research (e.g. is a jurisdiction needed? Does it need to be legally binding?)

Reviewer 2: > The community will have access to all recorded interviews

Will these interviews be anonymized? Regarding the example questions, I'd assume you're referring to treasury proposals rather than the grants program here, as for us (W3F Grants) it's a matter of compliance with the regulators, hence it doesn't make any sense to question it imho.

**Relevant related work.**

The majority of our team members have been integral members of Jur since its inception, and their expertise and experience are crucial to the success of this project. Here are some noteworthy milestones that align with the scope of our research:

- We were awarded and successfully completed a public tender by the Ministry of Justice (MoJ) through the LawtechUK initiative, where we served as the delivery partner for a feasibility study on the SME Dispute Resolution Platform. This collaboration involved researchers from Oxford University and Oxford Computer Consultants;

> - We developed and launched the Open Layer platform, which represents a pioneering game theory-based Online Dispute Resolution (ODR) system tailored for micro-claims;

Reviewer 2: Could you include a link? Also for the other products you've already launched.

- We have engaged in collaborative efforts with the United Kingdom Jurisdiction Taskforce (UKJT) to develop digital dispute resolution rules;

- We have participated in the leading legal tech acceleration program MDR Lab. This experience has allowed us to specifically focus on the area of digital arbitration.

These milestones exemplify our team's extensive involvement in the field, demonstrating our deep commitment to advancing the realms of jurisdiction, ODR, and legal technology.

**Intended venue for results publication and the timeline for publication.**

**What your project is not or will not provide or implement**

This research project is focused on conducting comprehensive investigations in the technical needs of the ecosystem, and as such, it will not involve a complete implementation of the open jurisdiction stack itself.

The ultimate deliverable of this project will be a go/no-go decision regarding the jurisdictional requirements within the ecosystem. It will provide valuable insights and recommendations based on the findings, enabling stakeholders to make informed decisions regarding the implementation of a jurisdiction framework.

> We also aim to release at least one open source component of the stack to engage and share with the community around how the development of the jurisdiction stack could look like. This endeavour will be done also in case of “no-go” outcome from the research. Sentiment and needs towards a jurisdiction may change in the future and we think it would be wise at least to leave a tangible outcome from this endeavour from where anyone can resume.

Reviewer 2: > This endeavour will be done also in case of “no-go” outcome from the research.

This sounds like a waste of resources tbh. Why would you implement something that doesn't work (if that was the case)?

Disclaimer: Jur has built an arbitration platform, the “Open Justice Platform”, that supports legally binding resolutions in 160+ jurisdictions. This platform could be one of the resolution methods available in the stack, alongside all the others that the community will see fit. In no way, this will be the suggested or required solution. It will be on the parties involved to choose the best resolution methods for their needs.

### Ecosystem Fit

**Help us locate your project in the Polkadot/Substrate/Kusama landscape and what problems it tries to solve by answering each of these questions:**

**Where and how does your project fit into the ecosystem?**

> Having actively participated in the Dotsama ecosystem over the past year and engaging with numerous stakeholders, we have gained valuable insights into the needs inherent to the ecosystem.

Reviewer 2: Could you give some specific examples, such as discussions/threads/blog articles/podcasts/conferences you've contributed to?

- Raffaele Battaglini ([link](https://scholar.google.com/citations?user=1wmRm_8AAAAJ&hl=it))

## Development Status :open_book:

In addition to our previous accomplishments, we have proactively initiated conversations with various stakeholders within the ecosystem, including representatives from Parity and Web3 Foundation as well as ecosystems' teams.

## Development Roadmap :nut_and_bolt:

### Overview

- **Estimated Duration:** 4 months
- **FTE:** 3 FTE
- **Costs:** 80,000 USD

The team involved has decided to work on a reduced market rate. If market rates were to be applied for professional or research activities, the cost would be prohibitively high. The team agrees to be paid on completion of each milestone.

### Milestone 1 - Data Collection, Stakeholder Interviews and First Draft

- **Estimated Duration:** 2 months
- **FTE:** 3 FTE
- **Costs:** 30,000 USDC

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| **0a** | **Copyrights and Licenses** | Apache 2.0 |

> | **0b** | **Documentation**           | NA |

Reviewer 2: What format/tech are you going to use to write the paper - latex? Or some kind of binary format?

| **0c** | **Methodology**             | We will provide a brief documentation of the methodology and process used for the research. This includes an explanation on why before proposing an actual scope of development we feel a thorough analysis phase is the better approach as this involves real world needs (e.g. legal enforcement) and it is not a purely technical delivery |

> | 1  | **Research**                | We will conduct research on a minimum of 10 past proposals and 10 grants, ecosystem prizes and future initiatives to gather some information around jurisdiction needs of the ecosystem. We think this sample should be enough to make informed decisions and make sure that the future technical delivery is supported by data.We will extract common data points, abstracting away the details, on which we will engage with stakeholders in the subsequent activity |

Reviewer 2: ```suggestion
| 1  | **Research**                | We will conduct research on a minimum of 10 past proposals and 10 grants, ecosystem prizes and future initiatives to gather some information around jurisdiction needs of the ecosystem. We think this sample should be enough to make informed decisions and make sure that the future technical delivery is supported by data. We will extract common data points, abstracting away the details, on which we will engage with stakeholders in the subsequent activity |
```

> | 2  | **Interviews**              | We will run a minimum of 20 interviews of about 45 minutes with relevant stakeholders, giving priority to the engagement around treasury proposals but also touching on the grants processes. All the interviews will be following a discussion guide format that aims at understanding the needs that need to be solved by the open jurisdiction stack.Examples of data points we will extract through those interviews are given in the section _The methodology that will be applied._ |

Reviewer 2: ```suggestion
| 2  | **Interviews**              | We will run a minimum of 20 interviews of about 45 minutes with relevant stakeholders, giving priority to the engagement around treasury proposals but also touching on the grants processes. All the interviews will be following a discussion guide format that aims at understanding the needs that need to be solved by the open jurisdiction stack. Examples of data points we will extract through those interviews are given in the section _The methodology that will be applied._ |
```

| 3  | **Shared Folder**           | We will share a folder with contents from interviews, aggregation of data points etc. basically anything that will be eventually part of the published paper |

### Milestone 2 - POC, Feedback and Publication

- **Estimated Duration:** 2,5 months
- **FTE:** 3 FTE
- **Costs:** 50,000 USDC

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| **0a** | **Copyrights and Licenses** | Apache 2.0 |
| **0b** | **Documentation**           | NA |

> | **0c** | **Methodology**             | We will provide a brief documentation of the methodology and process used for the researchThis builds up on the previous milestone. |

Reviewer 2: ```suggestion
| **0c** | **Methodology**             | We will provide a brief documentation of the methodology and process used for the research. This builds up on the previous milestone. |
```
